src/hooks/useTask.ts

```diff
@@ -8,37 +8,46 @@ export const useTask = () => {
   const [taskList, setTaskList] = useState<Task[]>([]);
 
   useEffect(() => {
-    taskData.getAllTaskData().then((task) => {
-      setTaskList([...task].reverse());
-    });
+    const allTaskData = async () => {
+      const data = await taskData.getAllTaskData();
+      setTaskList([...data].reverse());
+    };
+    allTaskData();
   }, []);
 
   // Taskの追加
   const addTaskListItem = (todoContent: string) => {
     //ulid()で一意のIdを取得
     const newTaskItem = { id: ulid(), task: todoContent, done: false };
 
-    taskData.addTaskData(newTaskItem).then((addTask) => {
-      setTaskList([addTask, ...taskList]);
-    });
+    const addTaskData = async () => {
+      const data = await taskData.addTaskData(newTaskItem);
+      setTaskList([data, ...taskList]);
+    };
+    addTaskData();
```

Comment:
ここの関数自体をasync関数にして直接awaitでOKです〜🙆‍♂️
```suggestion
    const data = await taskData.addTaskData(newTaskItem);
    setTaskList([data, ...taskList]);
```

おそらくuseEffect内の非同期処理の書き方に引きずられていそうです👀
useEffectに指定する関数はasyncにできないので、
useEffect内でasync関数を定義する
といったuseEffectの癖なので、通常の関数であれば上記で指摘したような書き方で大丈夫です🙆‍♂️